src/Umbraco.Core/Services/Implement/ContentService.cs

```diff
@@ -2201,7 +2201,7 @@ public IContent Copy(IContent content, int parentId, bool relateToOriginal, bool
                     while (page * pageSize < total)
                     {
                         var descendants = GetPagedDescendants(content.Id, page++, pageSize, out total);
-                        foreach (var descendant in descendants)
+                        foreach (var descendant in descendants.OrderBy(x => x.Level).ThenBy(y => y.SortOrder))
```

Comment:
This should be sorted via SQL/Database, this will not perform well.

Reply:
I tried passing "level" to the `GetPagedDescendants` method: 

`var descendants = GetPagedDescendants(content.Id, page++, pageSize, out total, ordering: Ordering.By("level"));`

But that doesn't work because we end up in `ContentRepositoryBase.ApplySystemOrdering` and `level` is not a supported field there. I don't know if it can "just" be added but even if we could then `Umbraco.Core.Services.Ordering` does not seem to support ordering on multiple columns so we wouldn't be able to construct the same ordering that being doing in Linq here.